doc/source/DoxygenLayout.xml

```diff
@@ -0,0 +1,226 @@
+<doxygenlayout version="1.0">
```

Comment:
does this work w/o doxygen? or is it needed for pybind11 mkdocs?

Reply:
it does work without doxygen but only because I disabled autodoc generation as I mentioned here https://github.com/BlueBrain/MorphIO/pull/248#discussion_r592468748. This PR is a placeholder only. It does not generated proper documentation from sources. Should I also include here autodoc generation as well?

Reply:
> This PR is a placeholder only

Yup, that's fine.

I think we should remove the doxygen stuff for now, and add it to the PR when proper documentation is generated.

Reply:
I made it but CI is failing due to unrelated reasons. I will try to rerun CI tomorrow.